src/main/java/core/basesyntax/storage/Storage.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.storage;
+
+import core.basesyntax.model.FruitBatch;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage implements StorageOperations {
+    private Map<String,FruitBatch> storage = new HashMap<>();
+
+    @Override
+    public boolean addFruits(FruitBatch fruitBatch) {
+        if (fruitBatch == null) {
+            throw new IllegalArgumentException("Cannot add null fruit batch");
+        }
+        String fruitName = fruitBatch.getFruitName();
+        FruitBatch storageBatch = storage.get(fruitName);
+        if (storageBatch != null
+                && (storageBatch.getBatchDay().isBefore(fruitBatch.getBatchDay())
```

Comment:
This `if` statement is too complex. Consider to simplify it or move to the separate method with `boolean` return type.